clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/artifact/ArtifactReplacerFactory.java

```diff
@@ -21,112 +21,96 @@
 import com.netflix.spinnaker.clouddriver.kubernetes.v2.artifact.ArtifactReplacer.Replacer;
 
 public class ArtifactReplacerFactory {
-  public static Replacer dockerImageReplacer() {
-    return Replacer.builder()
-        .replacePath(
-            "$..spec.template.spec['containers', 'initContainers'].[?( @.image == \"{%name%}\" )].image")
-        .findPath("$..spec.template.spec['containers', 'initContainers'].*.image")
-        .nameFromReference(
-            ref -> {
-              int atIndex = ref.indexOf('@');
-              // @ can only show up in image references denoting a digest
-              // https://github.com/docker/distribution/blob/95daa793b83a21656fe6c13e6d5cf1c3999108c7/reference/regexp.go#L70
-              if (atIndex >= 0) {
-                return ref.substring(0, atIndex);
-              }
+  public static final Replacer DOCKER_IMAGE =
```

Comment:
FYI I would have been inclined to make the constants `private` and kept the static methods since
1. the fact that they're singletons is not particularly relevant to clients
2. if you later need to add one that _isn't_ a singleton (perhaps it takes a parameter), the resulting static method would be more consistent if the rest of them were too

I generally try to limit my `public` constants to primitives. But whatever, definitely not a huge deal. I only really point it out  because everything else is too ✨ 💯  to comment on.